code/game/machinery/computer/cloning.dm

```diff
@@ -263,46 +278,53 @@
 				src.diskette.buf.types=DNA2_BUF_SE
 		src.diskette.name = "data disk - '[src.active_record.dna.real_name]'"
 		src.temp = "Save \[[href_list["save_disk"]]\] successful."
-		nanomanager.update_uis(src)
 
 	else if (href_list["refresh"])
 		nanomanager.update_uis(src)
+		
+	else if (href_list["selectpod"])
+		var/obj/machinery/clonepod/selected = locate(href_list["selectpod"])
+		if(istype(selected) && (selected in pods))
+			selected_pod = selected
 
 	else if (href_list["clone"])
 		var/datum/dna2/record/C = locate(href_list["clone"])
 		//Look for that player! They better be dead!
-		if(C)
+		if(istype(C))
 			//Can't clone without someone to clone.  Or a pod.  Or if the pod is busy. Or full of gibs.
-			if(!pod1)
+			if(!pods.len)
 				temp = "<span class=\"bad\">Error: No cloning pod detected.</span>"
-			else if(pod1.occupant)
-				temp = "<span class=\"bad\">Error: The cloning pod is currently occupied.</span>"
-			else if(pod1.biomass < CLONE_BIOMASS)
-				temp = "<span class=\"bad\">Error: Not enough biomass.</span>"
-			else if(pod1.mess)
-				temp = "<span class=\"bad\">Error: The cloning pod is malfunctioning.</span>"
-			else if(!config.revival_cloning)
-				temp = "<span class=\"bad\">Error: Unable to initiate cloning cycle.</span>"
-			else if(pod1.growclone(C))
-				temp = "Initiating cloning cycle..."
-				records.Remove(C)
-				qdel(C)
-				menu = 1
-			else
-				var/mob/selected = find_dead_player("[C.ckey]")
-				selected << 'sound/machines/chime.ogg'	//probably not the best sound but I think it's reasonable
-				var/answer = alert(selected,"Do you want to return to life?","Cloning","Yes","No")
-				if(answer != "No" && pod1.growclone(C))
+			else 
+				var/obj/machinery/clonepod/pod = selected_pod
+				if (!selected_pod)
+					temp = "<span class=\"bad\">Error: No cloning pod selected.</span>"
+				else if(pod.occupant)
+					temp = "<span class=\"bad\">Error: The cloning pod is currently occupied.</span>"
+				else if(pod.biomass < CLONE_BIOMASS)
+					temp = "<span class=\"bad\">Error: Not enough biomass.</span>"
+				else if(pod.mess)
+					temp = "<span class=\"bad\">Error: The cloning pod is malfunctioning.</span>"
+				else if(!config.revival_cloning)
+					temp = "<span class=\"bad\">Error: Unable to initiate cloning cycle.</span>"
+				else if(pod.growclone(C))
 					temp = "Initiating cloning cycle..."
 					records.Remove(C)
 					qdel(C)
 					menu = 1
 				else
-					temp = "Initiating cloning cycle...<br /><span class=\"bad\">Error: Post-initialisation failed. Cloning cycle aborted.</span>"
+					var/mob/selected = find_dead_player("[C.ckey]")
+					selected << 'sound/machines/chime.ogg'	//probably not the best sound but I think it's reasonable
```

Comment:
Nothing, really. It just... does nothing.

That being said, that bit of code is useless, seeing as pod.growclone(C) above it should already clone the selected ckey. For some reason there's this fallback in case growclone fails, in which case it for some reason asks the selected to be cloned and then... tries the same thing again. I'll add a check if the selected is actually selected, but it shouldn't change much.